src/ripple/app/tx/impl/AMMCreate.cpp

```diff
@@ -0,0 +1,347 @@
+    }();
+
+    // AMM account already exists (should not happen)
+    if (!ammAccount)
+    {
+        JLOG(j_.error()) << "AMM Instance: AMM already exists.";
+        return {ammAccount.error(), false};
+    }
+
+    // LP Token already exists. (should not happen)
+    auto const lptIss = ammLPTIssue(
+        amount.issue().currency, amount2.issue().currency, *ammAccount);
+    if (sb.read(keylet::line(*ammAccount, lptIss)))
+    {
+        JLOG(j_.error()) << "AMM Instance: LP Token already exists.";
+        return {tecDUPLICATE, false};
+    }
+
+    // Create AMM Root Account.
+    auto sleAMMRoot = std::make_shared<SLE>(keylet::account(*ammAccount));
+    sleAMMRoot->setAccountID(sfAccount, *ammAccount);
+    sleAMMRoot->setFieldAmount(sfBalance, STAmount{});
+    std::uint32_t const seqno{
+        ctx_.view().rules().enabled(featureDeletableAccounts)
+            ? ctx_.view().seq()
+            : 1};
+    sleAMMRoot->setFieldU32(sfSequence, seqno);
+    // Ignore reserves requirement, disable the master key, allow default
+    // rippling (AMM LPToken can be used as a token in another AMM, which must
+    // support payments and offer crossing), and enable deposit authorization to
+    // prevent payments into AMM.
+    // Note, that the trustlines created by AMM have 0 credit limit. This
+    // prevents shifting the balance between accounts via AMM. This is
+    // a desired behavior, though not an ideal solution; i.e. 0 means no
+    // trust except for AMMs where it means infinite trust.
+    sleAMMRoot->setFieldU32(
+        sfFlags, lsfAMM | lsfDisableMaster | lsfDefaultRipple | lsfDepositAuth);
+    sb.insert(sleAMMRoot);
+
+    // Calculate initial LPT balance.
+    auto const lpTokens = ammLPTokens(amount, amount2, lptIss);
+
+    // Create ltAMM
+    auto ammSle = std::make_shared<SLE>(ammKeylet);
+    ammSle->setFieldU16(sfTradingFee, ctx_.tx[sfTradingFee]);
+    ammSle->setAccountID(sfAMMAccount, *ammAccount);
+    ammSle->setFieldAmount(sfLPTokenBalance, lpTokens);
+    auto const& [issue1, issue2] = std::minmax(amount.issue(), amount2.issue());
+    ammSle->setFieldIssue(sfAsset, STIssue{sfAsset, issue1});
+    ammSle->setFieldIssue(sfAsset2, STIssue{sfAsset2, issue2});
+    // AMM creator gets the voting slot.
+    STArray voteSlots;
+    STObject voteEntry{sfVoteEntry};
+    voteEntry.setFieldU16(sfTradingFee, ctx_.tx[sfTradingFee]);
+    voteEntry.setFieldU32(sfVoteWeight, 100000);
+    voteEntry.setAccountID(sfAccount, account_);
+    voteSlots.push_back(voteEntry);
+    ammSle->setFieldArray(sfVoteSlots, voteSlots);
+    // AMM creator gets the auction slot for free.
+    auto& auctionSlot = ammSle->peekFieldObject(sfAuctionSlot);
+    auctionSlot.setAccountID(sfAccount, account_);
+    // current + sec in 24h
+    auto const expiration =
+        std::chrono::duration_cast<std::chrono::seconds>(
+            ctx_.view().info().parentCloseTime.time_since_epoch())
+            .count() +
+        24 * 3600;
```

Comment:
You use this same value in several other places. Maybe it should be more global. Only problem is I'm not sure where to put it. Maybe SystemParameters.h?

It's not like the number of seconds in a day is going to change, but having this same construct in so many places feels hinky.

Reply:
`AMMCore.h` works!